src/python/pants/backend/python/tasks2/pex_build_util.py

```diff
@@ -119,6 +125,44 @@ def dump_requirements(builder, interpreter, req_libs, log, platforms=None):
       locations.add(dist.location)
 
 
+def prepare_dist_workdir(dist_tgt, workdir, log):
+  """Prepare Python distribution directory for SetupPyRunner by copying the 
+  target sources into a working directory located in .pants.d. 
+
+  :param dist_target: The :class:`PythonDistribution` to prepare a directory for. 
+  :param workdir: The working directory for this task.
+  :param log: Use this logger.
+
+  """
+  # Make directory for local built distributions.
+  local_dists_workdir = os.path.join(workdir, 'local_dists')
+  if not os.path.exists(local_dists_workdir):
+    safe_mkdir(local_dists_workdir)
+
+  # Fingerprint distribution target and create subdirectory for that target.
+  fingerprint = dist_tgt.payload.fingerprint()
+  dist_target_dir = os.path.join(local_dists_workdir, fingerprint)
+  if not os.path.exists(dist_target_dir):
+    log.debug('Creating working directory for target %s with fingerprint %s', 
+      dist_tgt.name, fingerprint)
+    safe_mkdir(dist_target_dir)
+
+  # Copy sources and setup.py over for packaging.
+  sources_rel_to_target_base = dist_tgt.sources_relative_to_target_base()
```

Comment:
The codes for copying sources file should be taken care of with a invalidated block I believe. (https://github.com/pantsbuild/pants/blob/master/src/python/pants/task/task.py#L309)